Src/Microsoft.Dynamic/Generation/FieldBuilderExpression.cs

```diff
@@ -31,9 +31,19 @@ namespace Microsoft.Scripting.Generation {
     public class FieldBuilderExpression : Expression {
         private readonly FieldBuilder _builder;
 
+#if NETCOREAPP2_0
+        private readonly StrongBox<Type> _finishedType;
```

Comment:
Can we do this as a runtime check instead, e.g. make finishedType default to null and then check if its null in GetFieldInfo?